### DISCUSSION TITLE
技术选型问题

### DISCUSSION THREAD

**Discussion by hehos** [dream-num/univer#1731] [Q&A | 常见问题]
univer 不使用函数式+原型编程风格而用class语法风格是基于什么考虑呢，后者在性能上会不会不如前者呢？

**Comment by jikkai** [comment]
关于第一个问题在 discord  已经提到过了，简单来说是为了可读性 / 可维护性 / 生态支持，直接阅读我们的源码或许能比较深刻地了解到以上三点在开发中的比重。性能方面的话，由于不同的浏览器不同版本的内核会有不同程度的优化，导致表现方面可能存在些许差异，你可以寻找目标浏览器的性能测试报告来了解具体性能差异，这通常是微乎其微的。

**Reply by hehos** [reply]
在代码可读性和生态规问题上，建议源码用2个空格缩进，以保持跟主流的前端规范一致呢

**Reply by jikkai** [reply]
我个人的观点是：**合适**的总是优于**最好**的。
